### PR TITLE
Change integer handling with SSZ to use little endian

### PR DESCRIPTION
**Pull request by atoulme**
Applies to integer values, as well as list sizes.
Added direct support for UInt384.

### CODE REVIEW

File: bytes/src/main/java/net/consensys/cava/bytes/Bytes.java [hunk 1]
```diff
@@ -314,6 +314,79 @@ static Bytes ofUnsignedInt(long value) {
...
+  /**
+   * Return a 4 bytes little-endian encoded value corresponding to the provided value interpreted as an unsigned int
+   * value.
+   *
+   * @param value The value, which must fit an unsigned int.
+   * @return A 4 bytes value corresponding to {@code v}.
+   * @throws IllegalArgumentException if {@code v < 0} or {@code v} is too big to fit an unsigned 4-bytes int (that is,
+   *         if {@code v >= (1L << 32)}).
+   */
+  static Bytes ofUnsignedIntAsLittleEndian(long value) {
```
Comment (cleishm) [inline bytes/src/main/java/net/consensys/cava/bytes/Bytes.java]: `ofLittleEndianUnsignedInt`

Comment (cleishm) [reply bytes/src/main/java/net/consensys/cava/bytes/Bytes.java]: Or use a big/little param

File: bytes/src/main/java/net/consensys/cava/bytes/Bytes.java [hunk 2]
```diff
@@ -481,6 +554,29 @@ default int getInt(int i) {
...
   }
 
+  /**
+   * Retrieve the 4 bytes starting at the provided index in this value as an integer, using little endian.
+   *
+   * @param i The index from which to get the int, which must less than or equal to {@code size() - 4}.
+   * @return An integer whose value is the 4 bytes from this value starting at index {@code i}.
+   * @throws IndexOutOfBoundsException if {@code i &lt; 0} or {@code i &gt; size() - 4}.
+   */
+  default int getLittleEndianEncodedInt(int i) {
```
Comment (cleishm) [inline bytes/src/main/java/net/consensys/cava/bytes/Bytes.java]: `getLittleEndianInt`

File: bytes/src/main/java/net/consensys/cava/bytes/Bytes.java (hunk 1 above)
Comment (atoulme) [reply bytes/src/main/java/net/consensys/cava/bytes/Bytes.java]: maybe the param, as an overloaded method. big endian clearly should be default.

**Comment by atoulme** [comment]
Fixes issue #101

**Pull request merged**